packages/custom_lint/test/src/workspace_test.dart

```diff
@@ -724,25 +733,55 @@ void main() {
         );
       }
 
-      test('throws MissingPubspecError if package does not contain a pubspec',
+      test(
+        'finds pubspecs above analysis options file if there exists one',
+        () async {
+          final workspace = await createSimpleWorkspace(['package']);
+
+          final analysisFile = File(
+            p.join(workspace.path, 'package', 'analysis_options.yaml'),
+          );
+          analysisFile.createSync();
+          analysisFile.writeAsStringSync(analysisOptionsWithCustomLintEnabled);
+          final nestedAnalysisFile = File(
+            p.join(workspace.path, 'package', 'test', 'analysis_options.yaml'),
+          );
+          nestedAnalysisFile.createSync(recursive: true);
+          nestedAnalysisFile
+              .writeAsStringSync(analysisOptionsWithCustomLintEnabled);
+
+          final customLintWorkspace = await CustomLintWorkspace.fromPaths(
+            [p.join(workspace.path, 'package')],
+            workingDirectory: workspace,
+          );
+          // Expect one context root for the workspace and one for the test folder
+          expect(customLintWorkspace.contextRoots.length, equals(2));
+        },
```

Comment:
Could you remove those trailing commas and format again? Tests typically don't use them